spec/brainstem/api_docs/formatters/open_api_specification/version_2/endpoint_collection_formatter_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+require 'brainstem/api_docs/formatters/open_api_specification/version_2/endpoint_collection_formatter'
+require 'brainstem/api_docs/endpoint_collection'
+
+module Brainstem
+  module ApiDocs
+    module Formatters
+      module OpenApiSpecification
+        module Version2
+          describe EndpointCollectionFormatter do
+            let(:atlas)               { Object.new }
+            let(:endpoint_collection) { EndpointCollection.new(atlas) }
+            let(:nodoc)               { false }
+
+            subject { described_class.new(endpoint_collection) }
+
+            describe "#call" do
+              it "formats each endpoint" do
+                mock(subject).format_endpoints!
```

Comment:
I'm assuming that this is the easiest / most straightforward way to do this. Just want to register my complaint about mocking things on the object we're testing.

That can make it surprisingly easy to have totally broken code that's green, and perhaps indicates that a different structure would be "better".

Having said that, I know you're working within constraints and patterns were already there.

Reply:
I agree. I will add more integration tests within the same release if time permits or I will add it to the next release